components/trusted-app-mgt/org.wso2.carbon.identity.trusted.app.mgt/src/main/java/org/wso2/carbon/identity/trusted/app/mgt/services/TrustedAppMgtServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.trusted.app.mgt.services;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+import org.wso2.carbon.identity.application.common.model.TrustedApp;
+import org.wso2.carbon.identity.trusted.app.mgt.exceptions.TrustedAppMgtException;
+import org.wso2.carbon.identity.trusted.app.mgt.internal.TrustedAppMgtDataHolder;
+import org.wso2.carbon.identity.trusted.app.mgt.model.TrustedAndroidApp;
+import org.wso2.carbon.identity.trusted.app.mgt.model.TrustedIosApp;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.PlatformType;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.ANDROID_CREDENTIAL_PERMISSION;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.ANDROID_HANDLE_URLS_PERMISSION;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.IOS_CREDENTIAL_PERMISSION;
+
+/**
+ * Trusted App Management Service Implementation.
+ */
+public class TrustedAppMgtServiceImpl implements TrustedAppMgtService {
+
+    private static final Log LOG = LogFactory.getLog(TrustedAppMgtServiceImpl.class);
+
+    @Override
+    public List<TrustedAndroidApp> getTrustedAndroidApps() throws TrustedAppMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving trusted apps for Android platform.");
+        }
+        List<TrustedAndroidApp> trustedAndroidApps = new ArrayList<>();
+        try {
+            List<TrustedApp> trustedAppSet =
+                    TrustedAppMgtDataHolder.getInstance().getApplicationManagementService()
+                            .getTrustedApps(PlatformType.ANDROID);
+            for (TrustedApp trustedApp : trustedAppSet) {
+                TrustedAndroidApp trustedAndroidApp = new TrustedAndroidApp();
+                trustedAndroidApp.setPackageName(trustedApp.getAppIdentifier());
+                try {
+                    trustedAndroidApp.setThumbprints(resolveThumbprints(trustedApp));
+                    trustedAndroidApp.setPermissions(resolveAppPermissions(trustedApp));
+                } catch (TrustedAppMgtException e) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(e.getMessage() + "Hence removing the app from the trusted app list.");
```

Comment:
```suggestion
                        LOG.debug(e.getMessage() + "Hence removing the app from the trusted app list.", e);
```
better to make this a error log rather than a debug log

Reply:
This try catch block is removed and handled differently